server/src/com/thoughtworks/go/config/update/DeletePipelineConfigCommand.java

```diff
@@ -50,6 +50,22 @@ public void update(CruiseConfig cruiseConfig) throws Exception {
 
     @Override
     public boolean isValid(CruiseConfig preprocessedConfig) {
+        for (PipelineConfig pipeline : preprocessedConfig.getAllPipelineConfigs()) {
+            if(pipeline.materialConfigs().hasDependencyMaterial(pipelineConfig)){
+                Localizable.CurryableLocalizable message = LocalizedMessage.string("CANNOT_DELETE_PIPELINE_USED_AS_MATERIALS", pipelineConfig.name(), pipeline.name());
```

Comment:
replace `pipeline.name()` with `String.format("%s (%s)", pipeline.name(), pipeline.getOriginDisplayName())`
`pipelineConfig.name()` can remain as is since we do not allow deletion of remote(config-repo) pipelines. The check for that is currently missing at the controller side, the CRUD operations on remote pipelines will eventually fail deep down in the models, but we can fix that later.